pkg/nodediscovery/nodediscovery.go

```diff
@@ -525,6 +525,8 @@ func determineFirstInterfaceIndex(instanceType string) *int {
 		if defaults.IPAMPreAllocation > max {
 			idx = 0 // Include eth0
 		}
+	} else {
+		log.Warningf("Unable to find limits for instance type '%s', consider setting --update-ec2-adapter-limit-via-api=true", instanceType)
```

Comment:
Rather than using dynamic string formatting, let's use structured logging instead:

```
log.WithField("instance-type", instanceType).Warning(...)
```

Reply:
Actually, we might need string formatting to include the name of the option:

```
Warningf("Unable to find limits for instance type, consider setting --%s=true", option.UpdateEC2AdapterLimitViaAPI)
```

Reply:
absolutely, I splitted the change in 2 commits and updated accordingly 👌